packages/client/e2e/_utils/mpe-e2e-utils.ts

```diff
@@ -0,0 +1,642 @@
+                    withinMpeRoomsLibraryScreen('text="Your library"'),
+                );
+                await expect(libraryScreenTitle).toBeVisible();
+
+                const mpeRoomCard = otherUserPage.locator(
+                    withinMpeRoomsLibraryScreen(`text="${roomName}"`),
+                );
+                // eslint-disable-next-line no-constant-condition
+                while (true) {
+                    const roomHasBeenFound = await mpeRoomCard.isVisible();
+                    if (roomHasBeenFound === true) {
+                        break;
+                    }
+
+                    //This is tmp as the mpe room search machine is broken and only refresh on cancel
+                    const searchRoomInput = otherUserPage.locator(
+                        withinMpeRoomsLibraryScreen(
+                            'css=[placeholder^="Search a room"]',
+                        ),
+                    );
+                    await searchRoomInput.click();
+
+                    const cancelSearchRoomButton = otherUserPage.locator(
+                        withinMpeRoomsLibraryScreen('text="Cancel"'),
+                    );
+                    await cancelSearchRoomButton.click();
+                }
+            }),
+        );
+    }
+}
+
+/**
+ * Should be called from related mpe room view
+ * It then involves a redirection
+ * forcedDisconnectedPages should never contains a leaving user page
+ */
+
+interface NotRedirectedPage {
+    page: Page;
+    //Should assertion takes a page ? it could avoid future dev errors
+    assertion: () => Promise<void>;
+}
+export async function leaveMpeRoom({
+    leavingPage,
+    roomName,
+    forcedDisconnectedRedirectedOtherUsersPages,
+    forcedDisconnectedNotRedirectedOtherUsersPages,
+    redirectedLeavingUserPages,
+    notRedirectedLeavingUserPages,
+}: {
+    leavingPage: Page;
+    roomName: string;
+    forcedDisconnectedRedirectedOtherUsersPages?: Page[];
+    forcedDisconnectedNotRedirectedOtherUsersPages?: NotRedirectedPage[];
+    redirectedLeavingUserPages?: Page[];
+    notRedirectedLeavingUserPages?: NotRedirectedPage[];
+}): Promise<void> {
+    //Better ways to do that ? if undefined empty array ?
+
+    //Leaving user
+    notRedirectedLeavingUserPages = notRedirectedLeavingUserPages
+        ? notRedirectedLeavingUserPages
+        : [];
+
+    //In our testing leave method, leaving page is always redirected
+    redirectedLeavingUserPages = redirectedLeavingUserPages
+        ? [leavingPage, ...redirectedLeavingUserPages]
+        : [leavingPage];
+
+    //Other users
+    forcedDisconnectedRedirectedOtherUsersPages =
+        forcedDisconnectedRedirectedOtherUsersPages
+            ? forcedDisconnectedRedirectedOtherUsersPages
+            : [];
+
+    forcedDisconnectedNotRedirectedOtherUsersPages =
+        forcedDisconnectedNotRedirectedOtherUsersPages
+            ? forcedDisconnectedNotRedirectedOtherUsersPages
+            : [];
```

Comment:
```suggestion
export async function leaveMpeRoom({
    leavingPage,
    roomName,
    forcedDisconnectedRedirectedOtherUsersPages = [],
    forcedDisconnectedNotRedirectedOtherUsersPages = [],
    redirectedLeavingUserPages = [],
    notRedirectedLeavingUserPages = [],
}: {
    leavingPage: Page;
    roomName: string;
    forcedDisconnectedRedirectedOtherUsersPages?: Page[];
    forcedDisconnectedNotRedirectedOtherUsersPages?: NotRedirectedPage[];
    redirectedLeavingUserPages?: Page[];
    notRedirectedLeavingUserPages?: NotRedirectedPage[];
}): Promise<void> {
    // In our testing leave method, leaving page is always redirected
    const redirectedLeavingUserPagesWithLeavingPage = [leavingPage, ...redirectedLeavingUserPages];
```

Reply:
In my suggestion, we should be using `redirectedLeavingUserPagesWithLeavingPage` and not any more `redirectedLeavingUserPages` after we used it to compute `redirectedLeavingUserPagesWithLeavingPage`.

Reply:
```ts
    forcedDisconnectedRedirectedOtherUsersPages = [],
    forcedDisconnectedNotRedirectedOtherUsersPages = [],
    redirectedLeavingUserPages = [],
    notRedirectedLeavingUserPages = [],
```
You're so damn right, I was wondering what native option I've missed, there we are. Thanks

Checking deeper about the `redirectedLeavingUserPagesWithLeavingPage`